apps/list.c

```diff
@@ -637,15 +942,27 @@ static void list_pkey_meth(void)
     size_t i;
     size_t meth_count = EVP_PKEY_meth_get_count();
 
-    for (i = 0; i < meth_count; i++) {
-        const EVP_PKEY_METHOD *pmeth = EVP_PKEY_meth_get0(i);
-        int pkey_id, pkey_flags;
+    if (select_name == NULL) {
+        BIO_printf(bio_out, "Legacy:\n");
+        for (i = 0; i < meth_count; i++) {
+            const EVP_PKEY_METHOD *pmeth = EVP_PKEY_meth_get0(i);
+            int pkey_id, pkey_flags;
 
-        EVP_PKEY_meth_get0_info(&pkey_id, &pkey_flags, pmeth);
-        BIO_printf(bio_out, "%s\n", OBJ_nid2ln(pkey_id));
-        BIO_printf(bio_out, "\tType: %s Algorithm\n",
-                   pkey_flags & ASN1_PKEY_DYNAMIC ?  "External" : "Builtin");
+            EVP_PKEY_meth_get0_info(&pkey_id, &pkey_flags, pmeth);
+            BIO_printf(bio_out, " %s\n", OBJ_nid2ln(pkey_id));
+            BIO_printf(bio_out, "\tType: %s Algorithm\n",
+                       pkey_flags & ASN1_PKEY_DYNAMIC ?  "External" : "Builtin");
+        }
     }
+    BIO_printf(bio_out, "Provided:\n");
+    BIO_printf(bio_out, " Key encryption:\n");
+    list_asymciphers();
```

Comment:
s/Key encryption/Encryption/

Even though encrypting symmetric keys, that's not the only thing asymmetric encryption is used for, right?